src/test-unit/product.options.spec.js

```diff
@@ -0,0 +1,19 @@
+import { normalizeQueryStringParams } from '../lib/utils';
+const queryString = require('query-string');
+
+describe('Product Attributes', () => {
+    it('should normalize attribute parameters', () => {
+        const qs = 'attribute[0]=123&attribute[1]=456&qty=1';
+        const normalized = normalizeQueryStringParams(qs);
+
+        expect(normalized).toEqual(qs);
+    });
+
+    it('should remove empty array members', () => {
+        const qs = 'attribute[0]=123&attribute[1]=&qty=1';
+        const normalized = normalizeQueryStringParams(qs);
+        const parsed = queryString.parse(normalized, { arrayFormat: 'index' });
+       
+        expect(parsed['attribute'].length).toBe(1);
```

Comment:
Can you do the expect like this? instead of parsing the string. 
```js
expect(normalized).toEqual('attribute[0]=123&qty=1');
```

This way you are testing the values as well.